src/kernel/arch/x86/cmos.zig

```diff
@@ -0,0 +1,458 @@
+pub const RtcRegister = enum {
+    /// The seconds register
+    SECOND,
+
+    /// The minutes register
+    MINUTE,
+
+    /// The hours register
+    HOUR,
+
+    /// The days register
+    DAY,
+
+    /// The months register
+    MONTH,
+
+    /// The year register
+    YEAR,
+
+    /// The century register
+    CENTURY,
+};
+
+///
+/// Tell the CMOS chip to select the given register ready for read or writing to.
+///
+/// Arguments:
+///     IN reg: u8 - The register index to select in the CMOS chip.
+///
+inline fn selectRegister(reg: u8) void {
+    arch.outb(ADDRESS, reg);
+}
+
+///
+/// Tell the CMOS chip to select the given register ready for read or writing to. This also
+/// disables the NMI.
+///
+/// Arguments:
+///     IN reg: u8 - The register index to select in the CMOS chip.
+///
+inline fn selectRegisterNoNMI(reg: u8) void {
+    arch.outb(ADDRESS, reg | NMI_BIT);
+}
+
+///
+/// Write to the selected register to the CMOS chip.
+///
+/// Arguments:
+///     IN data: u8 - The data to write to the selected register.
+///
+inline fn writeRegister(data: u8) void {
+    arch.outb(DATA, data);
+}
+
+///
+/// Read the selected register from the CMOS chip.
+///
+/// Return: u8
+///     The value in the selected register.
+///
+inline fn readRegister() u8 {
+    return arch.inb(DATA);
+}
+
+///
+/// Select then read a register from the CMOS chip. This include a I/O wait to ensure the CMOS chip
+/// has time to select the register.
+///
+/// Arguments:
+///     IN reg: u8                    - The register index to select in the CMOS chip.
+///     IN comptime disable_nmi: bool - Whether to disable NMI when selecting a register.
+///
+/// Return: u8
+///     The value in the selected register.
+///
+inline fn readSelectedRegister(reg: u8, comptime disable_nmi: bool) u8 {
+    if (disable_nmi) {
+        selectRegisterNoNMI(reg);
+    } else {
+        selectRegister(reg);
+    }
+    arch.ioWait();
+    return readRegister();
+}
+
+///
+/// Select then write to a register to the CMOS chip. This include a I/O wait to ensure the CMOS chip
+/// has time to select the register.
+///
+/// Arguments:
+///     IN reg: u8                    - The register index to select in the CMOS chip.
+///     IN data: u8                   - The data to write to the selected register.
+///     IN comptime disable_nmi: bool - Whether to disable NMI when selecting a register.
+///
+inline fn writeSelectedRegister(reg: u8, data: u8, comptime disable_nmi: bool) void {
+    if (disable_nmi) {
+        selectRegisterNoNMI(reg);
+    } else {
+        selectRegister(reg);
+    }
+    arch.ioWait();
+    writeRegister(data);
+}
+
+///
+/// Read a register that corresponds to a real time clock register.
+///
+/// Arguments:
+///     IN reg: RtcRegister - A RTC register to select in the CMOS chip.
+///
+/// Return: u8
+///     The value in the selected register.
+///
+pub fn readRtcRegister(reg: RtcRegister) u8 {
```

Comment:
Taking an `RtcRegister` instance in the register read functions would make this unnecessary.